src/Domain/DomainCreators/AlignedBlocks.hpp

```diff
@@ -0,0 +1,93 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <limits>
+#include <vector>
+
+#include "DataStructures/Index.hpp"
+#include "Domain/Domain.hpp"
+#include "Domain/DomainCreators/DomainCreator.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/MakeArray.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace DomainCreators {
+
+/// \ingroup DomainCreatorsGroup
+/// \brief Create a Domain consisting of multiple aligned Blocks.
+///
+/// This is useful for setting up problems with piecewise smooth initial data.
```

Comment:
Maybe expand on how this is particularly more useful than a single element Block with multiple elements.